Change order of fields and field/title names on Cost Allocation and Other Funding + Budget and FFP pages. #2322

radavis opened the pull request:
Resolves #2322

### This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [ ] The change has been documented
  - [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate

### This feature is done when...

- [ ] Design has approved the experience
- [ ] Product has approved the experience

thetif reviewed (commented):
looks good

beparticular commented:
@radavis Thanks for all your hard work on this. Looking at the flow here, I think this is going to be a lot easier for people to understand. 

Couple of things on the export view - 
1) Under FFY 2021 Other Funding Amount, there is something that says "FFY Methodology". I think this might be a remnant of something, unless I am overlooking the corresponding text in the builder view. 
2) The line "Federal Financial Participation (FFP) and Cost Allocation" is still showing under each activity in the export view.

radavis commented on api/db/knex.js:
```diff
@@ -1,6 +1,13 @@
+  msg += "Terminating...";
+  console.error(msg);  /* eslint-disable-line no-console */
+  process.exit(1);
+}
```
Added this error condition around `NODE_ENV`, so that we don't continue executing if we don't know what knex configuration to use. This should help when troubleshooting.

radavis commented on api/knexrepl.js:
```diff
@@ -0,0 +1,17 @@
+const run = async () => {
+   r.context.knex = await knex;
+};
+run();
```
Added a `npm run knex-console` command, so that we can execute database queries using the knex SQL query builder.

radavis commented on api/package.json:
```diff
@@ -5,14 +5,18 @@
+    "seed-dev": "NODE_ENV=development DEV_DB_HOST=localhost npm run seed",
     "start": "node main.js",
-    "start-dev": "nodemon ./main.js -e js",
+    "start-dev": "NODE_ENV=development DEV_DB_HOST=localhost nodemon ./main.js -e js",
```
Added `migrate-dev` and `seed-dev` commands that call `migrate` and `seed` commands with the appropriate env vars when developing locally.

radavis commented on api/routes/apds/post.test.js:
```diff
@@ -64,6 +64,8 @@ tap.test('apds POST endpoint', async endpointTest => {
+    // Why is a 500 status being returned when the frontend sends malformed data?
+    // Should this not be a 4xx status? What the heck?
     tests.test(
       'sends a 500 if the newly-generated APD fails schema validation',
```
If the front-end sends bad data, I would expect the api to respond with a 4xx HTTP status code (client error), and not a 5xx HTTP status code (server error).

Looking at the rest of the API implementation, it seems like this was an error in judgement that was made early on, and we ran with it. I'm going to file a tech debt ticket for this, but likely we will not be able to address this until after ATO. Any thoughts on this, @thetif?

radavis commented:
Thank you for the feedback, @beparticular! I believe I have addressed all of the issues you have found. When you have the opportunity, please verify this on the preview deploy site.

beparticular reviewed (approved):
Issues look resolved. Thanks!
